### PR TITLE
[TS SDK] Add BCS types for effects

### PR DESCRIPTION
**Pull request by hayes-mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/typescript/src/bcs/bcs.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
+// Copyright (c) Mysten Labs, Inc.
```
Comment (hayes-mysten) [inline sdk/typescript/src/bcs/bcs.ts:1]: Everything in this files was just moved from bcs/index.ts

File: sdk/typescript/src/bcs/effects.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Mysten Labs, Inc.
```
Comment (hayes-mysten) [inline sdk/typescript/src/bcs/effects.ts:1]: These are all the new types required to parse effects bcs

File: sdk/typescript/src/bcs/types.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Mysten Labs, Inc.
```
Comment (hayes-mysten) [inline sdk/typescript/src/bcs/types.ts:1]: These types were all moved from bcs/index.ts

File: sdk/graphql-transport/src/mappers/transaction-block.ts [hunk 4]
```diff
@@ -353,3 +343,257 @@ function mapTransactionArgument(arg: typeof bcs.Argument.$inferType): SuiArgumen
...
+
+const OBJECT_DIGEST_DELETED = toB58(Uint8Array.from({ length: 32 }, () => 99));
+const OBJECT_DIGEST_WRAPPED = toB58(Uint8Array.from({ length: 32 }, () => 88));
+const OBJECT_DIGEST_ZERO = toB58(Uint8Array.from({ length: 32 }, () => 0));
+const ADDRESS_ZERO = normalizeSuiAddress('0x0');
+
+export function mapEffects(data: string): SuiTransactionBlockResponse['effects'] {
+	const effects = bcs.TransactionEffects.parse(fromB64(data));
+
+	let effectsV1 = effects.V1;
```
Comment (hayes-mysten) [inline sdk/graphql-transport/src/mappers/transaction-block.ts:355]: Mapping all this back to the JSON RPC representation is a bit involved.  We need to account for both V1 and V2 effects in bcs, and JSON RPC uses a separate effects v1 format that deviates a bit from bcs effects v1.  This function first creates effectsV1 either from bcs effects v1, or by converting v2 down to v1.  Then it has a separate mapping of the bcs v1 format to the JSON rpc version

File: sdk/graphql-transport/test/compatability.test.ts [hunk 5]
```diff
@@ -350,16 +350,18 @@ describe('GraphQL SuiClient compatibility', () => {
...
 			filter: {
 				FromAddress: toolbox.address(),
 			},
 			options: {
 				showBalanceChanges: true,
 				showEffects: true,
+				showRawEffects: true,
 				showEvents: true,
-				showInput: true,
+				// TODO inputs missing valueType
```
Comment (Jordan-Mysten) [inline sdk/graphql-transport/test/compatability.test.ts:363]: How big of a deal is this?

**Review by Jordan-Mysten — APPROVED** [review]
Generally looks good to me!

File: sdk/graphql-transport/test/compatability.test.ts (hunk 5 above)
Comment (hayes-mysten) [reply sdk/graphql-transport/test/compatability.test.ts:363]: This isn't a new issue in the graphql-transport.  Basically JSON rpc returns pure inputs like `{ type: 'pure', valueType: 'u64', value: 1 }` we currently return something like `{ type: 'pure', value: [1, 0, 0, 0, 0] }`.  I'm not sure how big a deal it is, but this is a known issue

**Pull request merged**